doc/classes/NavigationAgent.xml

```diff
@@ -0,0 +1,165 @@
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="distance_to_target" qualifiers="const">
+			<return type="float">
+			</return>
+			<description>
+				Returns the distance to the target location, using the agent's global position. The user must set the target location with [method set_target_location] in order for this to be accurate.
+			</description>
+		</method>
+		<method name="get_final_location">
+			<return type="Vector3">
+			</return>
+			<description>
+				Returns the reachable final location in global coordinates. This can change if the navigation path is altered in any way. Because of this, it would be best to check this each frame.
+			</description>
+		</method>
+		<method name="get_nav_path" qualifiers="const">
+			<return type="PoolVector3Array">
+			</return>
+			<description>
+				Returns the path from start to finish in global coordinates.
+			</description>
+		</method>
+		<method name="get_nav_path_index" qualifiers="const">
+			<return type="int">
+			</return>
+			<description>
+				Returns which index the agent is currently on in the navigation path's [PoolVector3Array].
+			</description>
+		</method>
+		<method name="get_navigation" qualifiers="const">
+			<return type="Node">
+			</return>
+			<description>
+				Returns the [Navigation] node that the agent is using for its navigation system.
+			</description>
+		</method>
+		<method name="get_next_location">
+			<return type="Vector3">
+			</return>
+			<description>
+				Returns a [Vector3] in global coordinates, that can be moved to, making sure that there are no static objects in the way. If the agent does not have a navigation path, it will return the origin of the agent's parent.
+			</description>
+		</method>
+		<method name="get_target_location" qualifiers="const">
+			<return type="Vector3">
+			</return>
+			<description>
+				Returns the user defined [Vector3] after setting the target location.
+			</description>
+		</method>
+		<method name="is_navigation_finished">
+			<return type="bool">
+			</return>
+			<description>
+				Returns true if the navigation path's final location has been reached.
+			</description>
+		</method>
+		<method name="is_target_reachable">
+			<return type="bool">
+			</return>
+			<description>
+				Returns true if the target location is reachable. The target location is set using [method set_target_location].
+			</description>
+		</method>
+		<method name="is_target_reached" qualifiers="const">
+			<return type="bool">
+			</return>
+			<description>
+				Returns true if the target location is reached. The target location is set using [method set_target_location]. It may not always be possible to reach the target location. It should always be possible to reach the final location though. See [method get_final_location].
+			</description>
+		</method>
+		<method name="set_navigation">
```

Comment:
Yes - but I did not do so because it was not a property at the time of the backport. And in current master, it is not apart of the class anymore due to moving things to a global setting in the project settings. Which I did not backport due to it not being agreed upon.

Reply:
I don't have a strong opinion. Properties are more idiomatic for the Godot API in general, but that's not a strong rule. The most important is to have both an easy upgrade path from the earlier API, and a new API that is as consistent as possible with the future one in 4.0. If this was moved somewhere else in 4.0, then it doesn't matter too much how it's done in 3.x as it won't impact 4.0+ APIs.